### PR TITLE
 [url_launcher] Adds macOS support

### PR DESCRIPTION
**Pull request by stuartmorgan**
## Description

Follows the structure established in #2119 to add a federated macOS implementation of the url_launcher plugin.

## Related Issues

Fixes macOS portion of https://github.com/flutter/flutter/issues/41721

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [ ] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [ ] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by stuartmorgan** [comment]
This doesn't include tests, since I'm not sure what the integration test strategy for federated plugins is. Should the example in the main plugin depend on the federated version and include macOS support?

File: packages/url_launcher/url_launcher_macos/ios/url_launcher_macos.podspec [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+                       DESC
+  s.homepage         = 'https://github.com/flutter/plugins/tree/master/packages/url_launcher/url_launcher_macos'
+  s.license          = { :file => '../LICENSE' }
+  s.author           = { 'Flutter Team' => 'flutter-dev@googlegroups.com' }
+  s.source           = { :path => '.' }
+  s.source_files = 'Classes/**/*'
+  s.public_header_files = 'Classes/**/*.h'
+  s.dependency 'Flutter'
+
+  s.ios.deployment_target = '8.0'
```
Comment (jaumard) [inline packages/url_launcher/url_launcher_macos/ios/url_launcher_macos.podspec]: Strange, we are under `/url_launcher_macos/ios/url_launcher_macos.podspec ` so iOS or MacOS ? If MacOS then  `s.ios.deployment_target` is wrong

Comment (stuartmorgan) [reply packages/url_launcher/url_launcher_macos/ios/url_launcher_macos.podspec]: See the summary/description entries.

(The ios/ folder is for iOS builds, regardless of the name of the package.)

**Comment by amirh** [comment]
> This doesn't include tests, since I'm not sure what the integration test strategy for federated plugins is. Should the example in the main plugin depend on the federated version and include macOS support?

Interesting point, we should probably think about it some more, I think it would be nice if the platform package could provide the example's platform scaffold (e.g for Android that will be the Activity, manifest etc...) and the e2e tests that are specified in the app-facing plugin will be executed as part of the macOS CI. Though devil is in the details, and I'm not sure how we'll end up with a pubspec that has all the example app's dependencies but also a dependency on the unendorsed implementation.

cc @collinjackson

File: packages/url_launcher/url_launcher_macos/README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# url\_launcher\_macos
+
+The macOS implementation of [`url_launcher`][1].
+
+## Usage
+
+To use this plugin in your Flutter Web app, simply add it as a dependency in
```
Comment (amirh) [inline packages/url_launcher/url_launcher_macos/README.md]: nit: MacOS

**Comment by collinjackson** [comment]
@franciscojma86 once we have driver testing of Mac apps on CI, this PR can land.

**Comment by stuartmorgan** [comment]
Obsoleted by https://github.com/flutter/plugins/pull/2383

**Pull request closed without merging**